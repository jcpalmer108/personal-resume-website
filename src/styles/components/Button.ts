import styled from "styled-components";
import { desktopBreakpoint, redstone, tabletBreakpoint } from "../variables";

export const Link = styled.a`
  width: 186px;
  height: 55px;
  display: flex;
  flex-direction: row;
  align-items: center;
  column-gap: 10px;
  background-color: ${redstone};
  justify-content: center;
  text-decoration: none;
  color: white;
  font-family: Gilroy-Medium;
  font-size: 12px;
  transition: all 500ms ease;

  @media (min-width: ${tabletBreakpoint}px) {
    width: 214px;
    height: 60px;
    font-size: 14px;
    border: 1px solid transparent;
    transition: all 500ms ease;

    &:hover {
      color: ${redstone};
      background: white;
      transition: all 500ms ease;
      font-size: 16px;
      border: ${redstone} 1px solid;

      img {
        filter: invert(59%) sepia(50%) saturate(1158%) hue-rotate(312deg)
          brightness(92%) contrast(95%);
        transition: all 500ms ease;
      }
    }
  }

  @media (min-width: ${desktopBreakpoint}px) {
    width: 218px;
    height: 64px;
  }
`;

export const ArrowRight = styled.img`
  width: 22px;
  padding-bottom: 3px;
`;
