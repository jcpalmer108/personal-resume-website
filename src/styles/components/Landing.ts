import styled from "styled-components"
import { desktopBreakpoint, tabletBreakpoint } from "../variables"

export const Main = styled.div`
  display: flex;
  `

export const Title = styled.h1`
  margin: 20px;
  width: 274px;

  @media (min-width: ${tabletBreakpoint}px) {
    width: 100%;
    margin: 0px;
  }

`

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  padding: 0px;

  
  
  @media (min-width: ${tabletBreakpoint}px) {
    text-align: left;
    padding: 25px;
    align-items: flex-start;
  }


  @media (min-width: ${desktopBreakpoint}px) {
    padding: 20px 60px;
  }
`

export const Description = styled.p`
  margin: 30px 20px;
  line-height: 20px;

  @media (min-width: ${tabletBreakpoint}px) {
    margin: 15px 0px 40px 0px;
    line-height: 27px;
  }

`

export const MobilePhoto = styled.img`
  height: 415px;
  width: 100%;
  object-fit: cover;
  object-position: 0px 92%;
  display: block;

  @media (min-width: ${tabletBreakpoint}px) {
    display: none;
  }
`

export const DesktopPhoto = styled.img`
  height: 563px;
  width: 367px;
  object-fit: cover;
  display: none;
  padding-right: 20px;

  @media (min-width: ${tabletBreakpoint}px) {
    display: block;
    padding-right: 25px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding-right: 60px;
  }

`

export const DiscoverMore = styled.div`
  height: 40px;
  visibility: hidden;
  display: flex;
  align-items: center;
  justify-content: center;
  column-gap: 10px;

  @media (min-width: ${tabletBreakpoint}px) {
    height: 65px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    height: 100px;
    visibility: visible;
  }

`

