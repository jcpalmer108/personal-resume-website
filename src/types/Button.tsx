export type ButtonProps = {
  label: string;
  url?: string;
  onClick?: Function;
};
