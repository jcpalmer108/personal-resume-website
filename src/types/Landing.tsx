import { Sections } from "./content";

export type LandingProps = {
  content: Sections | undefined;
};
