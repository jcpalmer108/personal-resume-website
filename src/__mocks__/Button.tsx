import React from "react";
import { ButtonProps } from "../types/Button";

export function Button({ label, url }: ButtonProps) {
  return (
    <div>
      {label}
      {url}
    </div>
  );
}
