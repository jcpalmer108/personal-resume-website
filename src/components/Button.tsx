import arrow from "../assets/images/icons/arrow-right.svg";
import { ArrowRight, Link } from "../styles/components/Button";
import { ButtonProps } from "../types/Button";

export default function Button({ label, url, onClick }: ButtonProps ) {
  if(!url && !onClick) {
    return null;
  }

  return (
    <div data-testid="Button" onClick={() => { if(onClick) onClick() }}>
      <Link href={url}>
        <span>{label.toUpperCase()}</span>
        <ArrowRight src={arrow} alt="arrow" />
      </Link>
    </div>
  )
}