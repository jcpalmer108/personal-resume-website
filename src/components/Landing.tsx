import React from "react";
import scroll from "../assets/images/icons/scroll.svg";
import {
  Content,
  Description,
  DesktopPhoto,
  DiscoverMore,
  Main,
  MobilePhoto,
  Title,
} from "../styles/components/Landing";
import { LandingProps } from "../types/Landing";
import Button from "./Button";
import Section from "./Section";
import SeparatingLine from "./SeparatingLine";

export default function Landing({ content }: LandingProps) {
  const defaultImage = "pexels-ekaterina-bolovtsova-4049791.jpg";

  return (
    <Section light wide>
      <Main data-testid="Main">
        <Content>
          <Title>{content?.title}</Title>
          <MobilePhoto
            src={require("../assets/images/gallery/" +
              (content?.image || defaultImage))}
            alt="landing mobile"
          />
          {content?.description &&
            content.description.map((paragraph, index) => (
              <Description
                key={`Description ${index + 1}`}
                data-testid={`Description ${index + 1}`}
              >
                {paragraph}
              </Description>
            ))}
          <Button
            label={content?.action?.text || "click me"}
            url={content?.action?.link || "#contact"}
          />
        </Content>
        <DesktopPhoto
          src={require("../assets/images/gallery/" +
            (content?.image || defaultImage))}
          alt="landing desktop"
        />
      </Main>
      <SeparatingLine light />
      <DiscoverMore data-testid="DiscoverMore">
        <img src={scroll} alt="scroll" />
        <p>Discover More</p>
      </DiscoverMore>
    </Section>
  );
}
