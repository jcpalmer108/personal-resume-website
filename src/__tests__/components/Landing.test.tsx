import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import Button from "../../components/Button";
import Landing from "../../components/Landing";

jest.mock("../../components/Button");

const params = {
  key: "landing",
  label: "Landing",
  inMenu: false,
  title: "test title",
  action: {
    text: "test action",
    link: "#contact",
  },
  image: "pexels-ekaterina-bolovtsova-4049791.jpg",
  description: ["test content 1", "test content 2"],
};

describe("Landing", () => {
  test("renders if only required params are passed in", () => {
    render(<Landing content={params} />);

    expect(screen.getByTestId("Main")).toMatchSnapshot();
    expect(screen.getByTestId("Main")).toHaveTextContent(params.title);
    params.description.forEach((item, index) => {
      expect(screen.getByTestId(`Description ${index + 1}`)).toBeTruthy();
      expect(screen.getByTestId(`Description ${index + 1}`)).toHaveTextContent(
        item
      );
    });

    expect(Button).toHaveBeenCalledTimes(1);
    expect(Button).toHaveBeenCalledWith(
      {
        label: params.action.text,
        url: params.action.link,
      },
      {}
    );
    expect(screen.getByRole("img", { name: "landing mobile" })).toBeTruthy();
    expect(screen.getByRole("img", { name: "landing mobile" })).toHaveAttribute(
      "src",
      params.image
    );
    expect(screen.queryByRole("img", { name: "landing desktop" })).toBeFalsy();
    expect(screen.getByTestId("DiscoverMore")).toMatchSnapshot();
    expect(screen.queryByRole("img", { name: "scroll" })).toBeFalsy();
  });
});
