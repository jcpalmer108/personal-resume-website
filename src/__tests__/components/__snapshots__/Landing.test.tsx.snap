// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders if only required params are passed in 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  margin: 20px;
  width: 274px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  padding: 0px;
}

.c4 {
  margin: 30px 20px;
  line-height: 20px;
}

.c3 {
  height: 415px;
  width: 100%;
  object-fit: cover;
  object-position: 0px 92%;
  display: block;
}

.c5 {
  height: 563px;
  width: 367px;
  object-fit: cover;
  display: none;
  padding-right: 20px;
}

@media (min-width:720px) {
  .c2 {
    width: 100%;
    margin: 0px;
  }
}

@media (min-width:720px) {
  .c1 {
    text-align: left;
    padding: 25px;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 20px 60px;
  }
}

@media (min-width:720px) {
  .c4 {
    margin: 15px 0px 40px 0px;
    line-height: 27px;
  }
}

@media (min-width:720px) {
  .c3 {
    display: none;
  }
}

@media (min-width:720px) {
  .c5 {
    display: block;
    padding-right: 25px;
  }
}

@media (min-width:1024px) {
  .c5 {
    padding-right: 60px;
  }
}

@media (min-width:1024px) {

}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<div
  class="c0"
  data-testid="Main"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      test title
    </h1>
    <img
      alt="landing mobile"
      class="c3"
      src="pexels-ekaterina-bolovtsova-4049791.jpg"
    />
    <p
      class="c4"
      data-testid="Description 1"
    >
      test content 1
    </p>
    <p
      class="c4"
      data-testid="Description 2"
    >
      test content 2
    </p>
  </div>
  <img
    alt="landing desktop"
    class="c5"
    src="pexels-ekaterina-bolovtsova-4049791.jpg"
  />
</div>
`;

exports[`Landing renders if only required params are passed in 2`] = `
.c0 {
  height: 40px;
  visibility: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-column-gap: 10px;
  column-gap: 10px;
}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {
  .c0 {
    height: 100px;
    visibility: visible;
  }
}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:1024px) {

}

<div
  class="c0"
  data-testid="DiscoverMore"
>
  <img
    alt="scroll"
    src="scroll.svg"
  />
  <p>
    Discover More
  </p>
</div>
`;
